garden_ai/app/pipeline.py

```diff
@@ -363,101 +366,64 @@ def register(
+        "--rm",
+        help="When set, the downloaded container image will be removed automatically",
     ),
 ):
-    import os
     import subprocess
-    import tempfile
-    import sys
 
+    client = GardenClient()
     try:
-        # Create a virtual environment in the temp directory
-        temp_dir = os.path.join(os.path.sep, tempfile.gettempdir(), env_name)
-        print(f"Setting up environment in {temp_dir} ...")
-        subprocess.run(["python3", "-m", "venv", temp_dir])
-
-        # Activate the environment os dependent
-        if sys.platform == "win32":
-            activate_script = os.path.join(temp_dir, "Scripts", "activate.bat")
-        elif sys.platform == "darwin":
-            activate_script = os.path.join(temp_dir, "bin", "activate")
-        else:
-            activate_script = os.path.join(temp_dir, "bin", "activate")
-
-        subprocess.run(f"source {activate_script}", shell=True)
-
-        # Upgrade pip in the virtual environment quietly
-        subprocess.run(
-            f"{temp_dir}/bin/python3 -m pip install -q --upgrade pip",
-            check=True,
-            shell=True,
-        )
-
-        # Install the requirements with nice spinner
-        with Progress(
-            SpinnerColumn(),
-            TextColumn("[progress.description]{task.description}"),
-            transient=True,
-        ) as progress:
-            progress.add_task(description="Installing requirements...", total=None)
-
-            subprocess.run(
-                f"{temp_dir}/bin/pip install -q -r {requirements_file}",
-                check=True,
-                shell=True,
+        if not doi and not container_uuid:
+            console.log(
+                "Either the DOI of a local pipeline or a container's UUID must be specified"
+            )
+            raise typer.Exit(code=1)
+        if doi and not container_uuid:
+            registered = client.get_registered_pipeline(doi)
+            cached_uuid = _read_local_cache().get(
+                get_cache_tag(
+                    registered.pip_dependencies,
+                    registered.conda_dependencies,
+                    registered.python_version,
+                )
             )
+            if cached_uuid is None:
+                console.log(
+                    "Fatal: the specified pipeline does not have its container uuid in the cache"
+                )
+                raise typer.Exit(code=1)
 
-        # Start Python shell in the virtual environment with the pipeline file
-        print("Starting Garden test shell. Loading your pipeline one moment...")
-        python_command = os.path.join(temp_dir, "bin", "python")
-        subprocess.run([python_command, "-i", pipeline_file])
+        container_info = client.compute_client.get_container(
+            container_uuid or cached_uuid, "docker"
+        )
+        location = container_info["location"]
+        location = location[location.index("bengal1") :]  # remove docker.io domain
```

Comment:
do we need to remove the docker.io domain? I thought that `docker pull 'docker.io/bengal1/bunchanumbers:latest'` would be the same as `docker pull 'bengal1/bunchanumbers:latest'` by default

Reply:
if we do need/want to strip it, it's better to hardcode the docker.io part we want to strip than ben g's username, since it's more likely that we change the user than the registry itself